sunxi: OPP table, cpu voltage control and thermal zone for Orange Pi PC

uzer00 opened the pull request:
Temperature info:
 cat /sys/class/thermal/thermal_zone0/temp
Scaling governors info (performance or powersave):
 cat /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor
 echo "powersave" > /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor

Warning! For hard working system active cooling must be

Tests resulte with stress [menuconfig>tools>stress]:
Temp after test with passive cooling (aluminum heat sink) about 78-82C
Idle temp about 30-36C

>\#!/bin/sh
\#Test script
for i in 480000 816000 1008000 1296000 1440000 ; do
 echo "cur temp: \`cat /sys/class/thermal/thermal_zone0/temp\`"
 echo "set freq to: " $i
 echo $i >/sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq
 stress -t 300 --cpu 4 --vm 2 --vm-bytes 512MB --io 4
 echo "temp after test: \`cat /sys/class/thermal/thermal_zone0/temp\`"
 sleep 60
done

Signed-off-by: Vitalij Alshevsky <v_alshevsky@tut.by>

Menion2k commented:
Already proposed in PR #825

hauke commented:
Are these the patches that are currently being proposed for upstream here: https://lkml.org/lkml/2018/2/5/900

uzer00 commented:
Yes, the patch based on Icenowy Zheng patches with some setups. I have only Orange Pi PC board for tests.

diizzyy commented:
With the pending including of Linux kernel 4.19 it would perhaps be a more streamlined process bumping sunxi to 4.19 as soon as possible?

diizzyy commented:
@uzer00 
I think we can close this and focus on 4.19 instead

Commit:
switch to 4.19

Commit:
sunxi: Add OPP table, cpu voltage control, cpu thermal sensor for Xunlong Orange Pi PC

Temperature info:
 cat /sys/class/thermal/thermal_zone0/temp
Scaling governors info (performance or powersave):
 cat /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor
 echo "powersave" > /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor

Warning! For hard working system active cooling must be

Tests resulte with stress [menuconfig>tools>stress]:
Temp after test with passive cooling (aluminum heat sink) about 78-82C
Idle temp about 30-36C

for i in 648000 816000 1008000 1344000 1440000 ; do
 echo "cur temp: `cat /sys/class/thermal/thermal_zone0/temp`"
 echo "set freq to: " $i
 echo $i >/sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq
 stress -t 300 --cpu 4 --vm 2 --vm-bytes 512MB --io 4
 echo "temp after test: `cat /sys/class/thermal/thermal_zone0/temp`"
 sleep 60
done

Signed-off-by: Vitalij Alshevsky <v_alshevsky@tut.by>

uzer00 commented:
@diizzyy 
Tested and work fine

ynezz commented:
Please see https://github.com/openwrt/openwrt/pull/1236#issuecomment-475022818, maybe you could unite the effort? Seems like you're working towards similar objective.

ynezz commented on target/linux/sunxi/Makefile:
```diff
@@ -14,7 +14,7 @@ FEATURES:=fpu usb ext4 display rtc squashfs
 MAINTAINER:=Zoltan HERPAI <wigyori@uid0.hu>
 
-KERNEL_PATCHVER:=4.14
+KERNEL_PATCHVER:=4.19
```
Please don't do this, it's probably going to be changed for all targets at once 19.03 is branched.

ynezz commented:
Please re-open the PR once you find a time (or interest) to work on this.
